presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -849,4 +969,61 @@ public FunctionMap(FunctionMap map, Iterable<? extends SqlFunction> functions)
             return functions.get(name);
         }
     }
+
+    private class ApplicableFunction
+    {
+        private final Signature declaredSignature;
+        private final Signature boundSignature;
+
+        private ApplicableFunction(Signature declaredSignature, Signature boundSignature)
+        {
+            this.declaredSignature = declaredSignature;
+            this.boundSignature = boundSignature;
+        }
+
+        public Signature getDeclaredSignature()
+        {
+            return declaredSignature;
+        }
+
+        public Signature getBoundSignature()
+        {
+            return boundSignature;
+        }
+
+        /**
+         * One method is more specific than another if invocation handled by the first method could be passed on to the other one
+         */
+        public boolean isMoreSpecificThan(ApplicableFunction other)
```

Comment:
This method doesn't seem to belong here. There's nothing special about it needing to operate on an "ApplicableFunction". All it cares about is the bound signature and the other's declared signature.